NebulaWorld/MonoBehaviours/Local/ChatManager.cs

```diff
@@ -0,0 +1,132 @@
+                    }
+                    else
+                    {
+                        Log.Debug($"Chat message is only sent locally");
+                    }
+                    chatBox.text = "";
+                }
+                else
+                {
+                    if (!chatBox.isFocused && Input.GetKeyDown(KeyCode.Return))
+                        chatBox.ActivateInputField();
+                }
+                if (!chatBox.isFocused)
+                {
+                    if (Input.GetKeyDown(KeyCode.Space))
+                    {
+                        SendMessageToChat($"[{DateTime.Now:HH:mm}] [system] : Space key pressed", ChatMessageType.SystemMessage);
+                    }
+                }
+            }
+
+            SendPlanetInfoMessage();
+        }
+
+        private void SendPlanetInfoMessage()
+        {
+            if (_sentLocation)
+                return;
+            if (GameMain.localPlanet == null && _attemptsToGetLocationCountDown-- > 0)
+            {
+                return;
+            }
+            var locationStr = GameMain.localPlanet == null ? "In Space" : GameMain.localPlanet.displayName;
+            Multiplayer.Session.Network.SendPacket(new NewChatMessagePacket(ChatMessageType.SystemMessage,
+                $"Connected, current location {locationStr}"));
+            _sentLocation = true;
+        }
+
+        public void SendMessageToChat(string text, ChatMessageType messageType)
+        {
+            if (messages.Count > maxMessages)
+            {
+                Destroy(messages[0].textObject.gameObject);
+                messages.Remove(messages[0]);
+            }
+
+            var newMsg = new Message { text = text };
+            GameObject nextText = Instantiate(textObject, chatPanel.transform);
+            newMsg.textObject = nextText.GetComponent<Text>();
+            newMsg.textObject.text = newMsg.text;
+            newMsg.textObject.color = MessageTypeColor(messageType);
+            Log.Info($"Adding message: {messageType} {newMsg.text}");
+            messages.Add(newMsg);
+            // alert user of new chat if panel closed
+            if (!chatWindow.activeSelf)
+            {
+                notifier.SetActive(true);
+            }
+        }
+
+        Color MessageTypeColor(ChatMessageType messageType)
+        {
+            Color color = Color.white;
+            switch (messageType)
+            {
+                case ChatMessageType.PlayerMessage:
+                    color = playerMessage;
+                    break;
+                case ChatMessageType.SystemMessage:
+                    color = info;
+                    break;
+            }
+
+            return color;
+        }
+
+        public void Toggle()
+        {
+            // ignore backtick if typing message
+            if (Input.GetKeyDown("`") && chatBox.isFocused)
```

Comment:
Seems like it would be good to use the `CustomKeyBindSystem` from CommonAPI for this, but I didn't want to start bringing in new dependencies without at least a little discussion